packages/react-search-ui/react-search-ui.d.ts

```diff
@@ -0,0 +1,345 @@
+    setFilter?: func;
+    a11yNotify?: func;
+  }
+
+  export class Facet extends Component<FacetProps> {}
+
+  interface ResultProps {
+    // Props
+
+    className?: string;
+    clickThroughTags?: string[];
+    titleField?: string;
+    urlField?: string;
+    view?: renderFunc;
+    result: ResultT;
+    shouldTrackClickThrough?: boolean;
+
+    // Actions
+
+    trackClickThrough?: func;
+  }
+
+  export class Result extends Component<ResultProps> {}
+
+  interface ResultsContainerProps {
+    // Props
+
+    className?: string;
+    clickThroughTags?: string[];
+    resultView?: renderFunc;
+    titleField?: string;
+    urlField?: string;
+    view?: renderFunc;
+    shouldTrackClickThrough?: boolean;
+
+    // State
+
+    results: ResultT[];
+  }
+
+  export class Results extends Component<ResultsContainerProps> {}
+
+  interface SearchBoxProps {
+    // Props
+
+    autocompleteMinimumCharacters?: number;
+    autocompleteResults?: boolean | {
+    clickThroughTags?: string[];
+    linkTarget?: string;
+    sectionTitle?: string;
+    shouldTrackClickThrough?: boolean;
+    titleField: string;
+    urlField: string
+    };
+    autocompleteSuggestions?: boolean | AutocompleteSection | {[key: string]: AutocompleteSection};
+    autocompleteView?: renderFunc;
+    className?: string;
+    shouldClearFilters?: boolean;
+    debounceLength?: number;
+    inputProps?: Partial<HTMLInputElement>;
+    inputView?: renderFunc;
+    onSelectAutocomplete?: func;
+    onSubmit?: func;
+    searchAsYouType?: boolean;
+    view?: renderFunc;
+
+    // State
+
+    autocompletedResults?: ResultT[];
+    autocompletedSuggestions?: { [key: string]: Suggestion[] };
+    searchTerm?: string;
+
+    // Actions
+
+    setSearchTerm?: func;
+    trackAutocompleteClickThrough?: func;
+  }
+
+  export class SearchBox extends Component<SearchBoxProps> {}
+
+  interface PagingInfoProps {
+    // Props
+
+    className?: string;
+    view?: renderFunc;
+
+    // State
+
+    pagingStart?: number;
+    pagingEnd?: number;
+    resultSearchTerm?: string;
+    totalResults?: number;
+  }
+
+  export class PagingInfo extends Component<PagingInfoProps> {}
+
+  interface PagingProps {
+    // Props
+
+    className?: string;
+    view?: renderFunc;
+
+    // State
+
+    current?: number;
+    resultsPerPage?: number;
+    totalPages?: number;
+
+    // Action
+
+    setCurrent?: func;
+  }
+
+  export class Paging extends Component<PagingProps> {}
+
+  interface ResultsPerPageProps {
+    // Props
+
+    className?: string;
+    view?: renderFunc;
+    options?: number[];
+
+    // State
+
+    resultsPerPage?: number;
+
+    // Actions
+
+    setResultsPerPage?: func;
+  }
+
+  export class ResultsPerPage extends Component<ResultsPerPageProps> {}
+
+  interface SortingProps {
+    // Props
+
+    className?: string;
+    label?: string;
+    sortOptions: SortOption[];
+    view?: renderFunc;
+
+    // State
+
+    sortDirection?: "asc" | "desc" | "";
```

Comment:
realized that this could be `SortDirection | ""`. Also had a general question: what's the difference between an undefined sort direction and one equal to ""?